test/common/upstream/zone_aware_load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,70 @@
+#include "zone_aware_load_balancer_fuzz_base.h"
+
+#include "test/mocks/upstream/host_set.h"
+
+namespace Envoy {
+namespace Upstream {
+
+void ZoneAwareLoadBalancerFuzzBase::initializeASingleHostSet(
+    const test::common::upstream::SetupPriorityLevel& setup_priority_level,
+    const uint8_t priority_level, uint16_t& port) {
+  LoadBalancerFuzzBase::initializeASingleHostSet(setup_priority_level, priority_level, port);
+  // Update local priority set if it exists - will mean load balancer is zone aware and has decided
+  // to construct local priority set
+  if (priority_level == 0 && local_priority_set_.get() != nullptr) {
+    MockHostSet& host_set = *priority_set_.getMockHostSet(priority_level);
+    local_priority_set_->getMockHostSet(0)->hosts_ = host_set.hosts_;
```

Comment:
is `local_priority_set_` always the 0th priority of priority_set_, or can it be a distinct host set with p=0?
Looking at: https://github.com/envoyproxy/envoy/blob/0d15fee3fcd0f32e4dd9240a3a42d909e3fae073/source/common/upstream/cluster_manager_impl.cc#L1034
Does it use the local cluster's priority set at p=0? Is this local cluster's priority set always equal to `priority_set_`?

Reply:
Yeah, I thought about this some, good catch. So, the local_priority_set_ is a subset of the 0th priority set, at least from the unit tests. Thus, for brevity and simplicity I decided to make it the 0th priority of priority_set_. Rethinking about this, this is probably to much of a shortcut. Let me know what you think it should be. If only I had more than two weeks left...

Reply:
I asked about this in Slack and got radio silence. Let's leave this as an open ended question and a TODO.

Reply:
I looked into this, local_priority_set_ is the local cluster's priority set (local cluster is where the Envoy is running, i.e. the originatig cluster). The load balancer's priority_set is the upstream cluster's priority set.

The difference between these two sizes determines where traffic goes, so it is worth adding in some logic eventually.

Seehttps://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/zone_aware

Reply:
Interesting, will add. Perhaps I can do this in my least request pr currently out.